src/components/section_view.rs

```diff
@@ -0,0 +1,140 @@
+use maud::{html, Markup};
+
+use crate::{
+    middlewares::SelectedCourses,
+    scraper::{Course, MeetingTime, Term, ThinCourse},
+};
+
+pub fn section_card(course: &Course, selected: &SelectedCourses, term: &Term) -> Markup {
+    let thin_course = ThinCourse {
+        subject_code: course.subject_code.to_owned(),
+        course_code: course.course_code.to_owned(),
+    };
+
+    let selected_crns = selected
+        .courses
+        .get(&thin_course)
+        .expect("course not found in selections?")
+        .iter()
+        .map(|s| s.crn)
+        .collect::<Vec<_>>();
+
+    html! {
+        form class="px-3" {
+            input type="hidden" name="course_code" value=(&course.course_code);
+
+            input type="hidden" name="subject_code" value=(&course.subject_code);
+
+            ul class="flex flex-col gap-3" {
+
+                @for section in &course.sections {
+                    @let checked = selected_crns.iter().any(|&c| c == section.crn);
+
+                    li class="flex flex-col text-white" {
+
+                        hr class="my-3 w-full"{}
+
+                        div class="px-3 flex flex-col gap-3 items-start justify-center" {
+
+                            div class="flex gap-3 justify-start items-center"{
+
+                                label class="cursor-pointer flex gap-2 justify-start items-center" {
+
+                                    @let url = format!("/term/{}/section", term);
+                                    (checkbox(url.as_str(), section.crn, checked))
+
+                                    "Section: " (&section.sequence_code)
+                                }
+
+                            }
+
+                            h4 class="" {
+                                "CRN: "(section.crn)
+                            }
+
+                            (meeting_time(&section.meeting_times))
+
+                            div class="flex justify-between items-center w-full" {
+
+                                span class="text-sm" {
+                                    "Seats: " (section.enrollment)"/"(section.enrollment_capacity)
+                                }
+
+                                span class="text-sm" {
+                                    "Waitlist: " (section.waitlist)"/"(section.waitlist_capacity)
```

Comment:
maybe only show waitlist count if the enrolment is full?

Reply:
but then you'd have blank white space in the section card for those which aren't full, it looks kinda imbalance imo